File: QuizApp2/app/src/main/java/com/example/quizapp/QuizQuestionActivity.java
package com.example.quizapp;

import android.content.Context;
import android.content.Intent;
import android.graphics.Color;
import android.graphics.Typeface;
import android.os.Bundle;
import android.util.Log;
import android.view.View;
import android.view.View.OnClickListener;
import android.widget.Button;
import android.widget.ImageView;
import android.widget.ProgressBar;
import android.widget.TextView;
import android.widget.Toast;
import androidx.appcompat.app.AppCompatActivity;
import androidx.core.content.ContextCompat;
import com.example.quizapp.R.id;
import java.util.ArrayList;
import java.util.Iterator;


public final class QuizQuestionActivity extends AppCompatActivity implements OnClickListener {
    private int mCurrentPosition = 1;
    private ArrayList mQuestionList;
    private int mSelectedOptionPosition;
    private int mCorrectAnswers;

    private String mUserName;

    protected void onCreate( Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        this.setContentView(R.layout.activity_quiz_question);

        mUserName = this.getIntent().getStringExtra("username");

        Constants constant=new Constants();
        mQuestionList = constant.getQuestions();

        ((TextView)findViewById(R.id.tv_option_one)).setOnClickListener((OnClickListener)this);
        ((TextView)findViewById(R.id.tv_option_two)).setOnClickListener((OnClickListener)this);
        ((TextView)findViewById(R.id.tv_option_three)).setOnClickListener((OnClickListener)this);
        ((TextView)findViewById(R.id.tv_option_four)).setOnClickListener((OnClickListener)this);
        ((Button)findViewById(R.id.btn_submit)).setOnClickListener((OnClickListener)this);

        setQuestion();
    }

    public final void setQuestion() {
        defualtOptionView();
        Question quesiton= (Question)mQuestionList.get(mCurrentPosition-1);

        ProgressBar progressBar = (ProgressBar)findViewById(id.progressBar);

        progressBar.setProgress(mCurrentPosition);

        TextView tv_progress = (TextView)findViewById(id.tv_progress);

        tv_progress.setText((mCurrentPosition+"/"+progressBar.getMax()).toString());

        TextView tv_question = (TextView)findViewById(id.tv_question);

        tv_question.setText(quesiton.getQuestion());

        ((ImageView)findViewById(id.tv_image)).setImageResource(quesiton.getImage());

        TextView tv_option_one = (TextView)findViewById(id.tv_option_one);

        tv_option_one.setText(quesiton.getOptionOne());

        TextView tv_option_two = (TextView)findViewById(id.tv_option_two);

        tv_option_two.setText(quesiton.getOptionTwo());

        TextView tv_option_three = (TextView)findViewById(id.tv_option_three);

        tv_option_three.setText(quesiton.getOptionThree());

        TextView tv_option_four = (TextView)findViewById(id.tv_option_four);

        tv_option_four.setText(quesiton.getOptionFour());

        /*Button var7=(Button)findViewById(id.btn_submit);
        if (mCurrentPosition == mQuestionList.size()) {
            var7.setText("Finish");
        } else {
            var7.setText("SUBMIT");
        }*/

    }

    private final void defualtOptionView() {
        ArrayList options = new ArrayList();
        options.add(0, (TextView)findViewById(R.id.tv_option_one));
        options.add(1, (TextView)findViewById(R.id.tv_option_two));
        options.add(2, (TextView)findViewById(R.id.tv_option_three));
        options.add(3, (TextView)findViewById(R.id.tv_option_four));
        Iterator var3 = options.iterator();
        while(var3.hasNext()) {
            TextView option = (TextView)var3.next();
            option.setTextColor(Color.parseColor("#7A8089"));
            option.setTypeface(Typeface.DEFAULT);
            option.setBackground(ContextCompat.getDrawable((Context)this,  R.drawable.default_option_border_bg));
        }

    }

    private final void selectedOptionView(TextView tv, int selectedOptionNum) {
        this.defualtOptionView();
        this.mSelectedOptionPosition = selectedOptionNum;
        tv.setTextColor(Color.parseColor("#363A43"));
        tv.setTypeface(tv.getTypeface(), Typeface.BOLD);
        tv.setBackground(ContextCompat.getDrawable((Context)this, R.drawable.selected_option_border_bg));
    }

    private final void answerView(int answer, int drawable) {
        TextView var10000;
        switch(answer) {
            case 1:
                TextView tv_option_one = (TextView)findViewById(R.id.tv_option_one);

                tv_option_one.setBackground(ContextCompat.getDrawable(this, drawable));
                break;
            case 2:
                TextView tv_option_two = (TextView)findViewById(R.id.tv_option_two);

                tv_option_two.setBackground(ContextCompat.getDrawable(this, drawable));
                break;
            case 3:
                TextView  tv_option_three = (TextView)findViewById(R.id.tv_option_three);

                tv_option_three.setBackground(ContextCompat.getDrawable(this, drawable));
                break;
            case 4:
                TextView tv_option_four = (TextView)findViewById(R.id.tv_option_four);

                tv_option_four.setBackground(ContextCompat.getDrawable(this, drawable));
        }

    }

    public void onClick(View v) {
        Integer var2 = v != null ? v.getId() : null;
        int var3 = R.id.tv_option_one;

        TextView var10;
        if (var2 != null) {

            if (var2 == var3) {
                var10 = (TextView)findViewById(R.id.tv_option_one);
                selectedOptionView(var10, 1);
                return;
            }
        }

        var3 =R.id.tv_option_two;

        if (var2 != null) {
            if (var2 == var3) {
                var10 = (TextView)findViewById(R.id.tv_option_two);

                this.selectedOptionView(var10, 2);
                return;
            }
        }

        var3 = R.id.tv_option_three;
        if (var2 != null) {
            if (var2 == var3) {
                var10 = (TextView)findViewById(R.id.tv_option_three);

                this.selectedOptionView(var10, 3);
                return;
            }
        }

        var3 = R.id.tv_option_four;
        if (var2 != null) {
            if (var2 == var3) {
                var10 = (TextView)findViewById(R.id.tv_option_four);

                this.selectedOptionView(var10, 4);
                return;
            }
        }

        var3 = id.btn_submit;
        if (var2 != null) {
            if (var2 == var3) {
                ArrayList var9;
                int var10000;
                int var10001;
                if (mSelectedOptionPosition == 0) {
                     mCurrentPosition++;

                    if (mCurrentPosition <= mQuestionList.size()) {
                        setQuestion();
                    } else {
                        Toast.makeText(this, "You have Successfully Completed Quiz", Toast.LENGTH_LONG).show();
                        Intent intent = new Intent(this, ResultActivity.class);
                        intent.putExtra("username", this.mUserName);
                        intent.putExtra("correct_answer", this.mCorrectAnswers);
                        intent.putExtra("total_questions", mQuestionList.size());
                        this.startActivity(intent);
                        this.finish();
                    }
                } else {

                    Question question = mQuestionList != null ? (Question)mQuestionList.get(this.mCurrentPosition - 1) : null;

                    if (question.getCorrectAnswer() != mSelectedOptionPosition) {
                        this.answerView(mSelectedOptionPosition,R.drawable.wrong1_option_border_bg);
                    } else {
                        mCorrectAnswers++;
                    }

                    this.answerView(question.getCorrectAnswer(), R.drawable.correct1_option_border_bg);



                    if (mCurrentPosition == mQuestionList.size()) {


                        ((Button)findViewById(R.id.btn_submit)).setText("Finish");
                    } else {

                        ( (Button)findViewById(R.id.btn_submit)).setText("Go to Next Question");
                    }

                 mSelectedOptionPosition = 0;
                }
            }
        }

    }
}

File: QuizApp/app/src/main/java/com/example/quizapp/QuizQuestionActivity.kt
package com.example.quizapp

import android.content.Intent
import android.content.res.ColorStateList
import android.graphics.Color
import android.graphics.Typeface
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.view.View
import android.widget.TextView
import android.widget.Toast
import androidx.core.content.ContextCompat
import kotlinx.android.synthetic.main.activity_quiz_question.*
import java.lang.reflect.Type
import kotlin.math.log

class QuizQuestionActivity : AppCompatActivity(),View.OnClickListener {
    private var mCurrentPosition:Int=1
    private var mQuestionList:ArrayList<Question>?=null
    private var mSelectedOptionPosition:Int=0
    private var mCorrectAnswers:Int=0
    private var mUserName:String?=null
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_quiz_question)
        mUserName=intent.getStringExtra(Constants.USER_NAME)
        mQuestionList=Constants.getQuestions()
        setQuestion()
        tv_option_one.setOnClickListener(this,)
        tv_option_two.setOnClickListener(this,)
        tv_option_three.setOnClickListener(this,)
        tv_option_four.setOnClickListener(this,)
        btn_submit.setOnClickListener(this,)

    }
    fun setQuestion()
    {
        defualtOptionView()
        val quesiton=mQuestionList!![mCurrentPosition-1]
        progressBar.progress=mCurrentPosition
        tv_progress.text="$mCurrentPosition/"+progressBar.max
        tv_question.text=quesiton.question
        tv_image.setImageResource(quesiton.image)
        tv_option_one.text=quesiton.optionOne
        tv_option_two.text=quesiton.optionTwo
        tv_option_three.text=quesiton.optionThree
        tv_option_four.text=quesiton.optionFour
        if(mCurrentPosition==mQuestionList!!.size)
        {
            btn_submit.text="Finish"
        }else{
            btn_submit.text="SUBMIT"
        }
    }

    private fun defualtOptionView(){

        val options = ArrayList<TextView>()
        options.add(0,tv_option_one)
        options.add(1,tv_option_two)
        options.add(2,tv_option_three)
        options.add(3,tv_option_four)
        for (option in options)
        {
            option.setTextColor(Color.parseColor("#7A8089"))
            option.typeface= Typeface.DEFAULT //default appearance of text
            option.background=ContextCompat.getDrawable(
                    this,
                    R.drawable.default_option_border_bg
            )
        }
    }
    private fun selectedOptionView(tv:TextView,selectedOptionNum:Int){
        defualtOptionView()
        mSelectedOptionPosition=selectedOptionNum
        tv.setTextColor(Color.parseColor("#363A43"))
        tv.setTypeface(tv.typeface,Typeface.BOLD) //default appearance of text
        tv.background=ContextCompat.getDrawable(
                this,
                R.drawable.selected_option_border_bg
        )
    }
    private fun answerView(answer:Int,drawable:Int)
    {
        when(answer)
        {
            1-> {
                tv_option_one.background=ContextCompat.getDrawable(this,drawable)
            }
            2-> {
                tv_option_two.background=ContextCompat.getDrawable(this,drawable)
            }
            3-> {
                tv_option_three.background=ContextCompat.getDrawable(this,drawable)
            }
            4-> {
                tv_option_four.background=ContextCompat.getDrawable(this,drawable)
            }
        }
    }
    override fun onClick(v: View?) {
        when(v?.id)
        {
            R.id.tv_option_one->{
                selectedOptionView(tv_option_one,1)
            }
            R.id.tv_option_two->{
                selectedOptionView(tv_option_two,2)
            }
            R.id.tv_option_three->{
                selectedOptionView(tv_option_three,3)
            }
            R.id.tv_option_four-> {
                selectedOptionView(tv_option_four, 4)
            }
            R.id.btn_submit-> {

                if(mSelectedOptionPosition==0){
                    mCurrentPosition++
                    when{
                        mCurrentPosition<=mQuestionList!!.size->{
                            setQuestion()
                        }
                        else->{
                            Toast.makeText(this@QuizQuestionActivity,"You have Successfully Completed Quiz",Toast.LENGTH_SHORT).show()
                            val intent=Intent(this,ResultActivity::class.java)
                            intent.putExtra(Constants.USER_NAME,mUserName)
                            intent.putExtra(Constants.CORRECT_ANSWERS,mCorrectAnswers)
                            intent.putExtra(Constants.TOTAL_Questions,mQuestionList!!.size)
                            startActivity(intent)
                            finish()
                            }

                        }

                }
                else{
                    val question=mQuestionList?.get(mCurrentPosition-1)
                    if(question!!.correctAnswer !=mSelectedOptionPosition)
                    {
                        answerView(mSelectedOptionPosition,R.drawable.wrong1_option_border_bg)
                    }else{
                        mCorrectAnswers++
                    }
                    answerView(question!!.correctAnswer,R.drawable.correct1_option_border_bg)
                    if(mCurrentPosition==mQuestionList!!.size)
                    {
                        btn_submit.text="Finish"
                    }else{
                        btn_submit.text="Go to Next Question"
                    }
                    mSelectedOptionPosition=0

                }

            }
            }
        }
    }



